src/containers/Template/Form/HSM/HSM.test.tsx

```diff
@@ -1,26 +1,46 @@
 import React from 'react';
-import { render, wait } from '@testing-library/react';
-import { BrowserRouter as Router } from 'react-router-dom';
+import { render, wait, within, fireEvent, cleanup } from '@testing-library/react';
 import { MockedProvider } from '@apollo/client/testing';
 
 import { HSM } from './HSM';
-import { Switch, Route } from 'react-router-dom';
-import { HSMList } from '../../List/HSMList/HSMList';
 import { TEMPLATE_MOCKS } from '../../Template.test.helper';
 
+afterEach(cleanup);
 const mocks = TEMPLATE_MOCKS;
 
 test('HSM form is loaded correctly in edit mode', async () => {
   const { getByText } = render(
     <MockedProvider mocks={mocks} addTypename={false}>
-      <Router>
-        <Switch>
-          <Route path="/template" exact component={HSMList} />
-        </Switch>
-        <HSM match={{ params: { id: 1 } }} />
-      </Router>
+      <HSM match={{ params: { id: 1 } }} />
     </MockedProvider>
   );
   await wait();
   expect(getByText('Edit HSM Template')).toBeInTheDocument();
 });
+
+test('check for validations for the HSM form', async () => {
+  const { getByText, container } = render(
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <HSM match={{ params: { id: null } }} />
+    </MockedProvider>
+  );
+  await wait();
+  expect(getByText('Add a new HSM Template')).toBeInTheDocument();
+
+  const { queryByText, getAllByText } = within(container.querySelector('form'));
+  const button = queryByText('Save');
+  fireEvent.click(button);
+  await wait();
+
+  // we should have 2 errors
+  expect(getAllByText('Required')).toHaveLength(2);
+
+  fireEvent.change(container.querySelector('input[name="label"]'), {
+    target: {
+      value: 'We are not allowing an really long title, and we should trigger validtaion for this.',
+    },
+  });
+
+  // we should still have 2 errors
+  expect(getAllByText('Required')).toHaveLength(2);
```

Comment:
will this test case change once we fix the error message that need to be displayed?

Reply:
Yes, this is a temporary fix as per current behavior.